challenges/2021-05-18-sum-of-squares/solutions/java/divyansh-ag-03/divyanshAg03.java

```diff
@@ -0,0 +1,54 @@
+class divyanshAg03
+{
+
+    static int getMinSquares(int n)
+    {
+        if (n <= 3) //ex: if n = 2, O/P = 1^2 + 1^2 = 2
+            return n;
+ 
+        // Create a dynamic programming
+        // to store squares
+        int dp[] = new int[n + 1];
+ 
+        // getMinSquares table for
+        // base case entries
+        dp[0] = 0;
+        dp[1] = 1;
+        dp[2] = 2;
+        dp[3] = 3;
+
+        // rest of table using recursion
+        for (int i = 4; i <= n; i++)
+        {
+         
+            // max value is i as i can
+            // always be represented
+            dp[i] = i;
+ 
+            // Going through all smaller numbers to recursively find minimum
```

Comment:
Please update or remove this comment (it's referring to the recursive approach from the first commit on this PR). Don't want to confuse folks who come later to learn from these examples.

Reply:
Thank you @ericwburden , It's done. Have a look at it.